### PR TITLE
Merge mysql-8.0.21 into Percona Server 8.0

### PR DESCRIPTION
**Pull request by kamil-holubicki**
1. Merge mysql 8.0.21 with conflicts
2. conflicts resolution + fixes (multiple commits)
3. Merge PS 8.0 head with conflicts
4. Resolve PS 8.0 head merge conflicts

### CODE REVIEW

**Comment by george-lorch** [comment]
5000+ files changed?!?

**Comment by kamil-holubicki** [comment]
> 5000+ files changed?!?

@georgelorch 
git diff mysql-8.0.20..mysql-8.0.21 --name-only | wc -l
10103

¯\\\_(ツ)\_/¯

**Comment by george-lorch** [comment]
Ahh, OK, I was thinking of Percona specific changes, not the .20...21 upstream merge itself

**Comment by kamil-holubicki** [comment]
Jenkins job: https://ps80.cd.percona.com/job/percona-server-8.0-param/715/
MTR results summary (Tab 05.08-2) https://docs.google.com/spreadsheets/d/1dDXVCtqpo_sJHVlb7mBc1YxeS1nG1BmzpTOiVl4oUK8/edit#gid=821969699

File: client/mysql.cc [hunk 1]
```diff
@@ -2658,17 +2658,7 @@ static char **new_mysql_completion(const char *text, int start, int end);
...
-||||||| merged common ancestors
-#if defined(USE_NEW_EDITLINE_INTERFACE)
-static int fake_magic_space(int, int);
-char *no_completion(const char *, int)
-#elif defined(USE_LIBEDIT_INTERFACE)
-static int fake_magic_space(int, int);
-=======
-#if defined(EDITLINE_HAVE_COMPLETION_CHAR)
->>>>>>> mysql-8.0.21
+#if defined(EDITLINE_HAVE_COMPLETION_CHAR) || defined(USE_NEW_XLINE_INTERFACE) || defined(USE_LIBEDIT_INTERFACE)
```
Comment (percona-ysorokin) [inline client/mysql.cc]: Please, resolve this the same was as in 5.7.31
https://github.com/percona/percona-server/commit/409b156ebf0ddfbcc7c845e96ce3fe2b62f40740

File: sql/auth/sql_authorization.cc [hunk 2]
```diff
@@ -1638,17 +1638,9 @@ class Get_access_maps : public boost::default_bfs_visitor {
...
-<<<<<<< HEAD
-        m_grantee{acl_user->user, strlen(acl_user->user),
-                  acl_user->host.get_host(), acl_user->host.get_host_len()},
-        m_effective_grants(effective_grants) {}
-||||||| merged common ancestors
-        m_grantee{acl_user->user, strlen(acl_user->user),
-                  acl_user->host.get_host(), acl_user->host.get_host_len()} {}
-=======
         m_grantee{acl_user->user, acl_user->get_username_length(),
                   acl_user->host.get_host(), acl_user->host.get_host_len()} {}
```
Comment (percona-ysorokin) [inline sql/auth/sql_authorization.cc]: Looks like a syntax error here. There should be no `{}`

File: sql/create_field.h [hunk 3]
```diff
@@ -181,20 +181,16 @@ class Create_field {
...
   bool is_array{false};
 
-<<<<<<< HEAD
   /*
     Store dict_id after verifying zip_dict_name exists. The stored id
     is filled in dd::Column::options and later used to fill TABLE_SHARE*
     zip_dict_name and zip_dict_data
-  */
+  */ 
   uint64_t zip_dict_id;
```
Comment (percona-ysorokin) [inline sql/create_field.h]: extra space

File: unittest/gunit/segfault-t.cc [hunk 4]
```diff
@@ -63,26 +63,10 @@ TEST_F(FatalSignalDeathTest, Segfault) {
    gtest library instead.
   */
   EXPECT_DEATH_IF_SUPPORTED(*pint = 42, "");
-<<<<<<< HEAD
-#elif defined(__SANITIZE_ADDRESS__)
+#elif defined(HAVE_ASAN)
   /* AddressSanitizer */
   EXPECT_DEATH_IF_SUPPORTED(*pint = 42, ".*ASAN:(DEADLYSIGNAL|SIGSEGV).*");
```
Comment (percona-ysorokin) [inline unittest/gunit/segfault-t.cc]: The conflict resolutions does not look right taking into account the fix from Oracle
https://github.com/mysql/mysql-server/commit/2ae0ca1ffe239e36995f4516290d8963ff434241
Please, double check with gcc-9/gcc-10 ASan

File: client/mysql.cc [hunk 5]
```diff
@@ -2706,7 +2706,7 @@ static void initialize_readline(char *name) {
   rl_completion_entry_function = &no_completion;
   rl_add_defun("magic-space", &fake_magic_space, -1);
 #else
-  rl_attempted_completion_function = (CPPFunction *)&new_mysql_completion;
+  rl_attempted_completion_function = reinterpret_cast<char** (*)(const char*, int, int)>(&new_mysql_completion);
```
Comment (percona-ysorokin) [inline client/mysql.cc]: Do the same as in 5.7.31

File: mysql-test/t/dd_schema_definition_debug.test [hunk 6]
```diff
@@ -343,21 +343,15 @@ INSERT INTO dd_published_schema
...
   VALUES('80019', 0,
     '6b98719841478028345c7c8696ef7fd2495ac06a145df6877b580ad2f319f6c7');
-
-||||||| merged common ancestors
-
-=======
 INSERT INTO dd_published_schema
   VALUES('80021', 0,
-    '80557e59b7af79e8a43e4b5efb7d5bab6a2db966935f0fe411b05b81cfdd1252');
+    '138bfe20f77905a83a20e1b665f91350c3278e3095eafd27dc310a2c7b051e04');
```
Comment (percona-ysorokin) [inline mysql-test/t/dd_schema_definition_debug.test]: This should be updated for case-insensitive filesystems as well

File: storage/innobase/fil/fil0crypt.cc [hunk 7]
```diff
@@ -1310,17 +1310,18 @@ struct rotate_thread_t {
...
       case SRV_SHUTDOWN_FLUSH_PHASE:
       case SRV_SHUTDOWN_LAST_PHASE:
       case SRV_SHUTDOWN_MASTER_STOP:
       case SRV_SHUTDOWN_RECOVERY_ROLLBACK:
-      case SRV_SHUTDOWN_PRE_DD_AND_SYSTEM_TRANSACTIONS:
-      case SRV_SHUTDOWN_PURGE:
-      case SRV_SHUTDOWN_DD:
         break;
     }
+    fprintf(stderr, "XXXXXXXXX %d\n", srv_shutdown_state.load());
```
Comment (percona-ysorokin) [inline storage/innobase/fil/fil0crypt.cc]: Looks like debug info. Is this intentional?

**Comment by percona-ysorokin** [comment]
Local run results (Ubuntu Focal, gcc-9, Debug):
```
[ 31%] main.pool_of_threads                     w8  [ fail ]  Found warnings/errors in error log file!
        Test ended at 2020-08-17 21:11:10
line
2020-08-17T18:11:02.873756Z 0 [Warning] [MY-013595] [Server] Failed to initialize TLS for channel: mysql_admin. See below for the description of exact issue.
2020-08-17T18:11:02.874100Z 0 [Warning] [MY-010069] [Server] Failed to set up SSL because of the following SSL library error: SSL context is not usable without certificate and private key
```

**Comment by percona-ysorokin** [comment]
```
[ 32%] binlog_57_decryption.binlog_rotate_key 'mix' w2  [ fail ]
        Test ended at 2020-08-17 21:13:42

CURRENT_TEST: binlog_57_decryption.binlog_rotate_key
mysqltest: At line 138: Query 'let $_eval_substmt_result= `$_eval_substmt_interp`' failed.
ERROR 1305 (42000): FUNCTION test.rotate_system_key does not exist
In included file ./include/eval.inc: 139
included from ./include/assert.inc: 61
included from /home/yura/ws/percona-server/mysql-test/suite/binlog_57_decryption/t/binlog_rotate_key.test: 8
```

**Comment by percona-ysorokin** [comment]
`innodb.missing_redologs` crashes with
```
/home/yura/ws/percona-build-8.0-debug_gcc9/runtime_output_directory/mysqld-debug(ut_dbg_assertion_failed(char const*, char const*, unsigned long)+0x1c5) [0x55edcd2f5253]
/home/yura/ws/percona-build-8.0-debug_gcc9/runtime_output_directory/mysqld-debug(Fil_shard::ibd_open_for_recovery(unsigned int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, fil_space_t*&)+0x341) [0x55edcd4d9307]
/home/yura/ws/percona-build-8.0-debug_gcc9/runtime_output_directory/mysqld-debug(Fil_system::ibd_open_for_recovery(unsigned int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, fil_space_t*&)+0xad) [0x55edcd4d9aa3]
/home/yura/ws/percona-build-8.0-debug_gcc9/runtime_output_directory/mysqld-debug(Fil_system::open_for_recovery(unsigned int)+0x158) [0x55edcd4e4a30]
/home/yura/ws/percona-build-8.0-debug_gcc9/runtime_output_directory/mysqld-debug(fil_tablespace_open_for_recovery(unsigned int)+0x23) [0x55edcd4e4bf3]
/home/yura/ws/percona-build-8.0-debug_gcc9/runtime_output_directory/mysqld-debug(recv_apply_hashed_log_recs(log_t&, bool)+0x274) [0x55edcd0caad1]
/home/yura/ws/percona-build-8.0-debug_gcc9/runtime_output_directory/mysqld-debug(srv_start(bool)+0x207b) [0x55edcd25bb6f]
/home/yura/ws/percona-build-8.0-debug_gcc9/runtime_output_directory/mysqld-debug(+0x4b5adb1) [0x55edccf52db1]
/home/yura/ws/percona-build-8.0-debug_gcc9/runtime_output_directory/mysqld-debug(+0x4b6e967) [0x55edccf66967]
/home/yura/ws/percona-build-8.0-debug_gcc9/runtime_output_directory/mysqld-debug(dd::bootstrap::DDSE_dict_init(THD*, dict_init_mode_t, unsigned int)+0xad) [0x55edccc75fe4]
/home/yura/ws/percona-build-8.0-debug_gcc9/runtime_output_directory/mysqld-debug(dd::upgrade_57::do_pre_checks_and_initialize_dd(THD*)+0x4c5) [0x55edccea4132]
```

**Comment by percona-ysorokin** [comment]
```
[ 35%] rocksdb.update_ignore 'write_prepared'   w2  [ fail ]
        Test ended at 2020-08-17 21:28:55

CURRENT_TEST: rocksdb.update_ignore
--- /home/yura/ws/percona-server/mysql-test/suite/rocksdb/r/update_ignore.result	2020-07-24 15:05:06.601961280 +0300
+++ /home/yura/ws/percona-build-8.0-debug_gcc9/mysql-test/var/2/log/update_ignore.reject	2020-08-17 21:28:55.228201032 +0300
@@ -34,7 +34,7 @@
 4	upd2a
 5	upd2a
 10000	upd2a
-1	a
+1	upd2a
 2	upd2a
 3	upd2a
 4	upd2a
```

**Comment by percona-ysorokin** [comment]
```
[ 41%] tokudb_parts.partition_alter1_1_tokudb   w10 [ fail ]
        Test ended at 2020-08-17 21:43:36

CURRENT_TEST: tokudb_parts.partition_alter1_1_tokudb
--- /home/yura/ws/percona-server/mysql-test/suite/tokudb_parts/r/partition_alter1_1_tokudb.result	2020-08-17 18:52:23.633300464 +0300
+++ /home/yura/ws/percona-build-8.0-debug_gcc9/mysql-test/var/10/log/partition_alter1_1_tokudb.reject	2020-08-17 21:43:35.625456726 +0300
@@ -59,7 +59,7 @@
 SELECT f_int1,f_int2,f_char1,f_char2,f_charbig FROM t0_template
 WHERE f_int1 BETWEEN 1 AND @max_row_div2 - 1;
 ALTER TABLE t1 ADD PRIMARY KEY(f_int2);
-ERROR HY000: A PRIMARY KEY must include all columns in the table's partitioning function
+ERROR HY000: A PRIMARY KEY must include all columns in the table's partitioning function (prefixed columns are not considered).
```

**Comment by percona-ysorokin** [comment]
```
[ 47%] tokudb_parts.partition_syntax_tokudb     w10 [ fail ]
        Test ended at 2020-08-17 21:54:22

CURRENT_TEST: tokudb_parts.partition_syntax_tokudb
--- /home/yura/ws/percona-server/mysql-test/suite/tokudb_parts/r/partition_syntax_tokudb.result	2020-08-17 18:52:23.729299582 +0300
+++ /home/yura/ws/percona-build-8.0-debug_gcc9/mysql-test/var/10/log/partition_syntax_tokudb.reject	2020-08-17 21:54:22.265554549 +0300
@@ -56,7 +56,7 @@
 PRIMARY KEY (f_int2)
 )
 PARTITION BY HASH(f_int1) PARTITIONS 2;
-ERROR HY000: A PRIMARY KEY must include all columns in the table's partitioning function
+ERROR HY000: A PRIMARY KEY must include all columns in the table's partitioning function (prefixed columns are not considered).
```

**Comment by percona-ysorokin** [comment]
```
[ 51%] innodb.check_ibd_filesize_16k            w12 [ fail ]
        Test ended at 2020-08-17 22:08:48

CURRENT_TEST: innodb.check_ibd_filesize_16k
--- /home/yura/ws/percona-server/mysql-test/suite/innodb/r/check_ibd_filesize_16k.result	2020-08-17 18:52:21.217322691 +0300
+++ /home/yura/ws/percona-build-8.0-debug_gcc9/mysql-test/var/12/log/check_ibd_filesize_16k.reject	2020-08-17 22:08:47.966984155 +0300
@@ -36,11 +36,11 @@
 Empty table:
 The size of file 'test/t3.ibd' is 7168 bytes.
 Rows inserted: 8
-The size of file 'test/t3.ibd' is 14336 bytes.
+The size of file 'test/t3.ibd' is 15360 bytes.
```

**Comment by percona-ysorokin** [comment]
```
[ 61%] keyring_vault.table_encrypt_kill         w6  [ fail ]
 CREATE TABLE tde_db.t_encrypt(c2 INT NOT NULL AUTO_INCREMENT PRIMARY KEY,
 c3 CHAR(255) Default 'No text',
-c4 JSON , 
+c4 JSON,
 c5 INT GENERATED ALWAYS AS (JSON_EXTRACT(c4,'$.key_a')) STORED,
 c6 INT GENERATED ALWAYS AS (JSON_EXTRACT(c4,'$.key_b')) VIRTUAL,
```

**Comment by percona-ysorokin** [comment]
```
[ 68%] rocksdb.delete_ignore 'write_prepared'   w9  [ fail ]
        Test ended at 2020-08-17 23:34:36

CURRENT_TEST: rocksdb.delete_ignore
--- /home/yura/ws/percona-server/mysql-test/suite/rocksdb/r/delete_ignore.result	2020-07-24 15:05:06.465962724 +0300
+++ /home/yura/ws/percona-build-8.0-debug_gcc9/mysql-test/var/9/log/delete_ignore.reject	2020-08-17 23:34:36.132131460 +0300
@@ -51,9 +51,6 @@
 Warning	1242	Subquery returns more than 1 row
 SELECT a,b FROM t1;
 a	b
-1	a
 SELECT c,d FROM t2;
 c	d
-foobar	10000
-foobar	10000
```

**Comment by percona-ysorokin** [comment]
```
[ 94%] tokudb.ext_key_1_tokudb                  w5  [ fail ]
        Test ended at 2020-08-18 00:37:15

CURRENT_TEST: tokudb.ext_key_1_tokudb
--- /home/yura/ws/percona-server/mysql-test/suite/tokudb/r/ext_key_1_tokudb.result	2020-08-17 18:52:23.241304070 +0300
+++ /home/yura/ws/percona-build-8.0-debug_gcc9/mysql-test/var/5/log/ext_key_1_tokudb.reject	2020-08-18 00:37:15.082084096 +0300
@@ -1,7 +1,7 @@
 drop table if exists t;
 select @@optimizer_switch;
 @@optimizer_switch
-index_merge=on,index_merge_union=on,index_merge_sort_union=on,index_merge_intersection=on,engine_condition_pushdown=on,index_condition_pushdown=on,mrr=on,mrr_cost_based=on,block_nested_loop=on,batched_key_access=off,materialization=on,semijoin=on,loosescan=on,firstmatch=on,duplicateweedout=on,subquery_materialization_cost_based=on,use_index_extensions=on,condition_fanout_filter=on,derived_merge=on,use_invisible_indexes=off,skip_scan=on,hash_join=on,favor_range_scan=off
+index_merge=on,index_merge_union=on,index_merge_sort_union=on,index_merge_intersection=on,engine_condition_pushdown=on,index_condition_pushdown=on,mrr=on,mrr_cost_based=on,block_nested_loop=on,batched_key_access=off,materialization=on,semijoin=on,loosescan=on,firstmatch=on,duplicateweedout=on,subquery_materialization_cost_based=on,use_index_extensions=on,condition_fanout_filter=on,derived_merge=on,use_invisible_indexes=off,skip_scan=on,hash_join=on,subquery_to_derived=off,prefer_ordering_index=on,favor_range_scan=off
```

**Comment by percona-ysorokin** [comment]
```
[ 94%] tokudb.ext_key_2_tokudb                  w5  [ fail ]
        Test ended at 2020-08-18 00:37:22

CURRENT_TEST: tokudb.ext_key_2_tokudb
--- /home/yura/ws/percona-server/mysql-test/suite/tokudb/r/ext_key_2_tokudb.result	2020-08-17 18:52:23.241304070 +0300
+++ /home/yura/ws/percona-build-8.0-debug_gcc9/mysql-test/var/5/log/ext_key_2_tokudb.reject	2020-08-18 00:37:22.037997876 +0300
@@ -1,7 +1,7 @@
 drop table if exists t;
 select @@optimizer_switch;
 @@optimizer_switch
-index_merge=on,index_merge_union=on,index_merge_sort_union=on,index_merge_intersection=on,engine_condition_pushdown=on,index_condition_pushdown=on,mrr=on,mrr_cost_based=on,block_nested_loop=on,batched_key_access=off,materialization=on,semijoin=on,loosescan=on,firstmatch=on,duplicateweedout=on,subquery_materialization_cost_based=on,use_index_extensions=on,condition_fanout_filter=on,derived_merge=on,use_invisible_indexes=off,skip_scan=on,hash_join=on,favor_range_scan=off
+index_merge=on,index_merge_union=on,index_merge_sort_union=on,index_merge_intersection=on,engine_condition_pushdown=on,index_condition_pushdown=on,mrr=on,mrr_cost_based=on,block_nested_loop=on,batched_key_access=off,materialization=on,semijoin=on,loosescan=on,firstmatch=on,duplicateweedout=on,subquery_materialization_cost_based=on,use_index_extensions=on,condition_fanout_filter=on,derived_merge=on,use_invisible_indexes=off,skip_scan=on,hash_join=on,subquery_to_derived=off,prefer_ordering_index=on,favor_range_scan=off
```

**Comment by percona-ysorokin** [comment]
```
[ 97%] tokudb.type_datetime                     w5  [ fail ]
        Test ended at 2020-08-18 00:45:58

CURRENT_TEST: tokudb.type_datetime
--- /home/yura/ws/percona-server/mysql-test/suite/tokudb/r/type_datetime.result	2020-08-17 18:52:23.317303371 +0300
+++ /home/yura/ws/percona-build-8.0-debug_gcc9/mysql-test/var/5/log/type_datetime.reject	2020-08-18 00:45:58.587624029 +0300
@@ -404,7 +404,7 @@
 2002-02-02
 Warnings:
 Warning	1287	Setting user variables within expressions is deprecated and will be removed in a future release. Consider alternatives: 'SET variable=expression, ...', or 'SELECT expression(s) INTO variables(s)'.
-Warning	1292	Incorrect date value: '' for column 'f1' at row 1
+Warning	1292	Incorrect datetime value: ''
```

**Comment by percona-ysorokin** [comment]
```
[ 98%] tokudb_bugs.commit_index_end_1           w6  [ fail ]
        Test ended at 2020-08-18 00:51:41

CURRENT_TEST: tokudb_bugs.commit_index_end_1
--- /home/yura/ws/percona-server/mysql-test/suite/tokudb_bugs/r/commit_index_end_1.result	2020-08-17 18:52:23.505301642 +0300
+++ /home/yura/ws/percona-build-8.0-debug_gcc9/mysql-test/var/6/log/commit_index_end_1.reject	2020-08-18 00:51:40.719420829 +0300
@@ -13,5 +13,5 @@
 3	SUBQUERY	t3	NULL	ref	c	c	15	const	1	100.00	Using index
 2	DEPENDENT SUBQUERY	t2	NULL	ALL	NULL	NULL	NULL	NULL	2	50.00	Using where
 Warnings:
-Note	1003	/* select#1 */ select `test`.`t1`.`a` AS `a` from `test`.`t1` where <in_optimizer>(`test`.`t1`.`a`,<exists>(/* select#2 */ select 1 from `test`.`t2` where (<cache>(`test`.`t1`.`a`) = `test`.`t2`.`b`)))
+Note	1003	/* select#1 */ select `test`.`t1`.`a` AS `a` from `test`.`t1` where <in_optimizer>(`test`.`t1`.`a`,<exists>(/* select#2 */ select `test`.`t2`.`b` from `test`.`t2` where (<cache>(`test`.`t1`.`a`) = `test`.`t2`.`b`)))
```

**Comment by percona-ysorokin** [comment]
In this PR we have about 20 commits with identical title `PS-7014: Merge MySQL 8.0.21` that will make it really hard to browse history in future.
Ideally, commit message titles should include brief description of actual changes as well.
E. g. `PS-7014 Merge MySQL 8.0.21: Fixed compilation for WITH_READLINE=system`

**Comment by percona-ysorokin** [comment]
Generally speaking, having commits with conflict markers is something every dev team should avoid.
We decided to make a compromise here and allow a single (merge with conflicts) commit in order to make reviewers life easier.
However, in my opinion, splitting (resolve conflicts) part into several commits is too much - they should be squashed into a single one.

**Comment by percona-ysorokin** [comment]
Moreover, having `Merge remote-tracking branch` commit is also against best practices - it just makes unnecessary noise in the commit history.
Rebasing would make this PR much clearer.

**Comment by percona-ysorokin** [comment]
One more comment regarding commit messages.
Please, use references to upstream/Percona issues and commits that are causing conflicts - this helps a lot with understanding why and how they are resolved.
I understand that this takes time but this is utterly important for the next person who will be doing another merge in a couple of months when they encounter a conflict in the same code part.
Here, you will be able to find a big number of examples
https://github.com/percona/percona-server/pull/3846

**Comment by percona-ysorokin** [comment]
Please, also add
```
leak:/usr/bin/zip
leak:/usr/bin/python3.8
```
to `mysql-test/lsan.supp`

**Comment by percona-ysorokin** [comment]
The following files still have merge conflict markers
```
./mysql-test/r/dd_is_compatibility_ci.result
./mysql-test/r/information_schema_ci.result
./mysql-test/r/mysqlshow_ci.result
./mysql-test/suite/funcs_1/r/is_key_column_usage_ci.result
./man/mysqldumpslow.1
./storage/innobase/buf/buf0flu.cc
./extra/libevent/libevent-2.1.11-stable/ChangeLog
```

**Comment by percona-ysorokin** [comment]
`./man/mysqldumpslow.1` must not be changed - it's a custom version with percona-specific info

**Comment by percona-ysorokin** [comment]
The following MTR test cases fail on case-insensitive data directory
```
main.information_schema_ci
main.dd_upgrade_error_ci
main.dd_is_compatibility_ci
funcs_1.is_key_column_usage_ci
main.mysqlshow_ci
```

**Comment by dutow** [comment]
@percona-ysorokin PR updated. I left the following failures as is:

* SSL tests - fail because the certificates expired. Also expired on upstream.
* innodb_sys_tablespace_encrypt_basic - fails on the current 8.0 branch, not related to the merge

**Comment by percona-ysorokin** [comment]
Another round of testing on 2020-08-21 (Ubuntu Focal gcc-9 Debug)
`encryption.keyring_and_mk_encryption_exclusions_d_e` [ fail ]
`binlog_57_decryption.binlog_rotate_key` [ fail ]
`rocksdb.update_ignore` [ fail ]
`tokudb.dir-per-db-with-custom-data-dir` [ fail ]
`tokudb_parts.partition_alter1_1_tokudb` [ fail ]
`keyring_vault.table_encrypt_kill` [ fail ]
`rocksdb.delete_ignore` [ fail ]
`sys_vars.innodb_sys_tablespace_encrypt_basic` [ fail ] (this is the one you mentioned)
`sys_vars.optimizer_switch_basic` [ fail ] (remaining merge conflict markers)
`tokudb.ext_key_1_innodb` [ fail ] (remaining merge conflict markers)
`tokudb.ext_key_2_innodb` [ fail ] (remaining merge conflict markers)

**Comment by percona-ysorokin** [comment]
`grep -rl . -e "||||||"`
`grep -rl . -e "<<<<<<"`
`grep -rl . -e ">>>>>>"`

The following files still have merge conflict markers
```
./mysql-test/r/dd_is_compatibility_ci.result
./mysql-test/r/information_schema_ci.result
./mysql-test/r/mysqlshow_ci.result
./mysql-test/suite/sys_vars/r/optimizer_switch_basic.result
./mysql-test/suite/tokudb/r/ext_key_2_innodb.result
./mysql-test/suite/tokudb/r/ext_key_1_innodb.result
./mysql-test/suite/funcs_1/r/is_key_column_usage_ci.result
./extra/libevent/libevent-2.1.11-stable/ChangeLog
./doc/source/tokudb/tokudb_intro.rst
```

**Comment by percona-ysorokin** [comment]
It looks like Marcelo's coredumper patch is now in conflict with with the
`shutdown_report` feature introduced in 8.0.21
in this patch
`Bug#30909369: MTR DOES NOT NOTICE A SERVER ABORT AFTER "BYE"`
https://github.com/mysql/mysql-server/commit/6a93f653

```
[100%] shutdown_report                          w0  [ fail ]
```

```
 - found '/home/yura/ws/percona-build-8.0-debug_gcc9/extra/coredumper'
 - found '/home/yura/ws/percona-build-8.0-debug_gcc9/extra/coredumper/src/coredumper.cmake'

Trying 'dbx' to get a backtrace

Trying 'gdb' to get a backtrace
 - found '/home/yura/ws/percona-build-8.0-debug_gcc9/extra/coredumper/src/CMakeFiles/Export/cmake/coredumper-debug.cmake'

Trying 'dbx' to get a backtrace

Trying 'gdb' to get a backtrace
 - found '/home/yura/ws/percona-build-8.0-debug_gcc9/extra/coredumper/src/CMakeFiles/Export/cmake/coredumper.cmake'

Trying 'dbx' to get a backtrace

Trying 'gdb' to get a backtrace
 - found '/home/yura/ws/percona-build-8.0-debug_gcc9/extra/coredumper/src/CMakeFiles/coredumper.dir'
 - found '/home/yura/ws/percona-build-8.0-debug_gcc9/extra/coredumper/src/CMakeFiles/coredumper.dir/coredumper.c.o'

Trying 'dbx' to get a backtrace

Trying 'gdb' to get a backtrace
```

Most probably because `coredumper` has the word `core` in it.

**Comment by dutow** [comment]
* binlog_57_decryption.binlog_rotate_key: existing failure on current 8.0 branch

**Comment by dutow** [comment]
Coredump issue is also unrelated and exists on current trunk.

**Comment by kamil-holubicki** [comment]
> `grep -rl . -e "||||||"`
> `grep -rl . -e "<<<<<<"`
> `grep -rl . -e ">>>>>>"`
> 
> The following files still have merge conflict markers
> 
> ```
> ./mysql-test/r/dd_is_compatibility_ci.result
> ./mysql-test/r/information_schema_ci.result
> ./mysql-test/r/mysqlshow_ci.result
> ./mysql-test/suite/sys_vars/r/optimizer_switch_basic.result
> ./mysql-test/suite/tokudb/r/ext_key_2_innodb.result
> ./mysql-test/suite/tokudb/r/ext_key_1_innodb.result
> ./mysql-test/suite/funcs_1/r/is_key_column_usage_ci.result
> ./extra/libevent/libevent-2.1.11-stable/ChangeLog
> ./doc/source/tokudb/tokudb_intro.rst
> ```

ChangeLog - this merge artifact is visible on upstream as well. I left it as it is.
All other fixed.

**Comment by percona-ysorokin** [comment]
@dutow `shutdown_report` is an MTR feature introduced in upstream 8.0.21 - not sure how it can fail in the current PS trunk.

In my opinion, after the merge, we've got a regression in the MTR's `shutdown_report` and this should be fixed as part of this merge PR.

**Comment by percona-ysorokin** [comment]
@dutow I could be wrong but it looks like extending the regexp in
https://github.com/percona/percona-server/blob/8.0/mysql-test/mysql-test-run.pl#L393
should fix the problem.

**Comment by percona-ysorokin** [comment]
@dutow Here are the results of the 2020-08-25 (commit 0357697) MTR run (Ubuntu Focal gcc-9 Debug)
`binlog_57_decryption.binlog_rotate_key` [ fail ] - existing 8.0 trunk issue
`rocksdb.update_ignore` [ fail ] - not re-recorded
`keyring_vault.table_encrypt_kill`  [ fail ] - not re-recorded after changed include file
`rocksdb.delete_ignore` [ fail ] - not-re-recorded
`main.dd_is_compatibility_cs` [ fail ] - a regression introduced in the latest commit (0357697)
`sys_vars.innodb_sys_tablespace_encrypt_basic` [ fail] - existing 8.0 trunk issue

**Comment by kamil-holubicki** [comment]
Hi @percona-ysorokin 
rocksdb.update_ignore and rocksdb.delete_ignore are related to https://bugs.mysql.com/bug.php?id=100352

**Comment by percona-ysorokin** [comment]
@kamil-holubicki 
Then may be we should update `rocksdb.update_ignore` and `rocksdb.delete_ignore` (change `a != ()` to `a NOT IN ()`).
This bug indeed seems to be an overlooked syntax validation rule - personally if I would never write anything like this.
Moreover, I am pretty sure the intention of those `rocksdb` tests was to test RocksDB functionality rather than validate syntax.
So, changing queries to proper unambiguous constructions is fine.

**Comment by kamil-holubicki** [comment]
Hi @percona-ysorokin , I agree that WHERE c < b OR a != ( SELECT 1 UNION SELECT 2 ); makes no sense, but please note that test name is delete_**ignore**, so in my opinion it is done intentionally. If we change the above statement to proper one WHERE c < b OR a NOT IN(1, 2) it will work, but there will be no trigger for IGNORE, so I'm afraid the test will test something else that was intended to test.
In 8.0.20 the behavior was that wrong part of WHERE (a!=SELECT 1 UNION SELECT 2) was ignored, so the result was as it was executed statement with only WHERE c < b + warning was issued.
I think this is what the author of this test wanted to achieve.

Bottom line:
Of course we can modify the test as proposed and re-record the result, but we will change test case. Moreover, current test discovered real bug in MySql, If it was written as proposed, the bug would stay undiscovered.

**Comment by percona-ysorokin** [comment]
@kamil-holubicki OK, let's leave these 2 tests as they are for now and monotor what Oracle decides regarding https://bugs.mysql.com/bug.php?id=100352.

So, there are 3 remaining issues:
1. `keyring_vault.table_encrypt_kill` [ fail ] - not re-recorded after changed include file
2. `main.dd_is_compatibility_cs` [ fail ] - a regression introduced in the latest commit (0357697)
3. `report_shutdown` and `coredumper` conflict

**Comment by kamil-holubicki** [comment]
@percona-ysorokin All 3 issues have been fixed

**Comment by percona-ysorokin** [comment]
@kamil-holubicki Still one remaining `// KH:` comment in `tokudb_backup.cc`
Moreover, the following comment is not addressed
https://github.com/percona/percona-server/pull/3978#discussion_r471824386

**Comment by kamil-holubicki** [comment]
@percona-ysorokin, 
1. Code cleaned up
2. Segfault test reverted to its upstream version as we discussed on Slack.

**Review by percona-ysorokin — APPROVED** [review]
LGTM
